src/Microsoft.SqlTools.Hosting/Utility/SqlClientEventListener.cs

```diff
@@ -36,10 +36,12 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         /// <param name="eventData">The data for the event</param>
         protected override void OnEventWritten(EventWrittenEventArgs eventData)
         {
-            if (eventData.Payload == null)
+            // Skip EventCounters as they can come from any event source and will pollute traces captured.
+            if (eventData.Payload == null || eventData.EventName.Equals(nameof(EventCounter)))
             {
                 return;
             }
+
             foreach (object payload in eventData.Payload)
```

Comment:
Is the payload a string in the cases we want to log? If so could we just check that here and only log it if it's a string? That would save us from having to do this again if a different object type is added as a payload (although I suppose that maybe we want to leave this since the objects added might have a ToString implementation and actually be useful...)

Reply:
It's commonly a string for events like 'Trace', 'Error' and depends on what the driver writes using `EventSource.WriteEvent`. I believe we could always expect/request a `ToString()` implementation from driver if that happens in future if we see something like that again, so that should be fine. These counters came from .NET 7, and that too polluting logs largely without we enabling them.. seemed a bit strange.